Userland/Libraries/LibGfx/Font/Emoji.cpp

```diff
@@ -46,14 +46,29 @@ Bitmap const* Emoji::emoji_for_code_points(ReadonlySpan<u32> const& code_points)
     return bitmap.ptr();
 }
 
+template<typename CodePointIterator>
+static bool could_be_emoji(CodePointIterator const& it)
+{
+    if (it.done())
+        return false;
+
+    static auto const emoji_property = Unicode::property_from_string("Emoji"sv);
+    if (!emoji_property.has_value()) {
+        // This means Unicode data generation is disabled. Always check the disk in that case.
+        return true;
+    }
+
+    return Unicode::code_point_has_property(*it, *emoji_property);
+}
+
 template<typename CodePointIterator>
 static Bitmap const* emoji_for_code_point_iterator_impl(CodePointIterator& it)
 {
     // NOTE: I'm sure this could be more efficient, e.g. by checking if each code point falls
     // into a certain range in the loop below (emojis, modifiers, variation selectors, ZWJ),
     // and bailing out early if not. Current worst case is 10 file lookups for any sequence of
     // code points (if the first glyph isn't part of the font in regular text rendering).
```

Comment:
Could this note also be updated/removed?
Or are there still more potential checks to be added.
(It's not a FIXME so leaving it isn't a big deal either)

Reply:
Yeah I left it for now because we can probably do more to bail early, not sure though